network/builder/src/builder.rs

```diff
@@ -418,6 +419,43 @@ impl NetworkBuilder {
         self
     }
 
+    fn add_configuration_change_listener(&mut self, role: RoleType) -> &mut Self {
+        let conn_mgr_reqs_tx = self
+            .conn_mgr_reqs_tx()
+            .expect("ConnectivityManager must be installed for validator");
+        let (simple_discovery_reconfig_subscription, simple_discovery_reconfig_rx) =
+            gen_simple_discovery_reconfig_subscription();
+        self.reconfig_subscriptions
+            .push(simple_discovery_reconfig_subscription);
+
+        self.configuration_change_listener_builder =
+            Some(ConfigurationChangeListenerBuilder::create(
+                role,
+                conn_mgr_reqs_tx,
+                simple_discovery_reconfig_rx,
+            ));
+        self.build_configuration_change_listener()
+            .start_configuration_change_listener()
```

Comment:
I think this is the only problem with this strategy, is cases like here, where we have 3 functions to separate functionality, but then, they're all called back to back anyways.

Reply:
This is still in the space of intermediate/prep steps.  The end result is that these three functions should not be called back-to-back, but that final separation cannot be done until all of the foundations are laid.

(See the endstate of #4626 for the separation)

Reply:
Note:  when this batch of changes goes throught (#4810, #4792, #4794), I'll be able to 
a) decouple the build/start calls within network_builder
b) remove additional unecessary fields from network_builder
c) tighten the network_builder::create logic a bit more
d) get the handling of *network* to be reasonable in main_node
e) start thinking about the other top level components (mempool, consensus, statesync)